### PR TITLE
8142 opacity ramp

### LINKED ISSUE #8142: Possibility to add general opacity if user styles by value (category or ramp)

### PR DESCRIPTION
**Pull request by ivanmalagon**
Related issue: #8142

Now the user can apply an opacity value on category or by value styling.

Points

8142-points

Polygons

8142-polygons

Lines

8142-lines

CC @xavijam @egsanchez 

Vamos!

### CODE REVIEW

**Comment by javierarce** [comment]
One general question: shouldn't the gradients and categories in the input field reflect the selected opacity value? (at least in the animated preview they appear to be 100% opaque)

**Comment by hacheka** [comment]
I asked @xavijam about that and he told me not to modify the square that previews the color since the opacity is not changing that particular color but all of them.

Anyway, it's true that it's a different behaviour from the fixed color picker so perhaps it's good that @urbanphes take a look on that.

File: lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-categories/input-color-categories.js [hunk 1]
```diff
@@ -178,9 +178,12 @@ module.exports = CoreView.extend({
       return { val: color, color: color, title: this.model.get('domain')[i] };
     }, this);
 
+    var opacity = typeof this.model.get('opacity') !== 'undefined' ? this.model.get('opacity') : 1;
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-categories/input-color-categories.js]: Option: !_.isUndefined(this.model.get('opacity'))

File: lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-ramps/input-ramp-content-view.js [hunk 2]
```diff
@@ -50,16 +50,22 @@ module.exports = CoreView.extend({
       };
     }, this);
 
+    var opacity = (typeof this.model.get('opacity') !== 'undefined') ? this.model.get('opacity') : 1;
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-ramps/input-ramp-content-view.js]: Same I said lines above.

**Comment by xavijam** [comment]
Hey, I've just realized what @javierarce said too, we should "preview" the opacity in the field when it is changed so:
screen shot 2016-12-05 at 14 47 50

Found in the acceptance tests, so please, let's change this 💋 .

**Comment by xavijam** [comment]
Also problems with the input alignment:

screen shot 2016-12-05 at 14 49 55

**Comment by xavijam** [comment]
Happening with categories too, so looks like a generic CSS problem:

screen shot 2016-12-05 at 14 52 50

File: app/assets/stylesheets/editor-3/color-picker.css.scss [hunk 3]
```diff
@@ -161,3 +161,52 @@
...
+
+.OpacityEditor-inputWrapper {
+  position: absolute;
+  bottom: 0;
+  right: 0;
+}
+
+.OpacityEditor-inputWrapper input {
+  height: 22px;
+}
```
Comment (xavijam) [inline app/assets/stylesheets/editor-3/color-picker.css.scss]: Hound rules! Please ^^^

File: app/assets/stylesheets/editor-3/color-picker.css.scss [hunk 4]
```diff
@@ -161,3 +161,52 @@
...
+
+.OpacityEditor-inputWrapper {
+  position: absolute;
+  right: 0;
+  bottom: 0;
+}
+
+.OpacityEditor-inputWrapper input {
+  height: 22px;
```
Comment (xavijam) [inline app/assets/stylesheets/editor-3/color-picker.css.scss]: check the line-height, it is likely that it is the origin of the CSS problem.

**Comment by xavijam** [comment]
Reviewed in staging (:+1:), passing frontend tests finally (:+1:), auto CR for the last things (:+1:), waiting last green for deploying ([ ]).

**Pull request merged**